File: benchmarks/server/src/sirannon_bench/core/registry.py
"""Map an engine name to its driver."""

from __future__ import annotations

from .config import Config
from .driver import Driver


def build_driver(engine: str, config: Config, durability: str) -> Driver:
    if engine == "sirannon":
        from ..drivers.sirannon import SirannonHttpDriver

        return SirannonHttpDriver(
            base_url=config.sirannon.base_url,
            database_id=config.sirannon.database_id,
            durability=durability,
            max_in_flight=config.max_in_flight,
        )
    if engine == "postgres":
        from ..drivers.postgres import PostgresDriver

        return PostgresDriver(config.postgres, durability)
    raise ValueError(f"unknown engine {engine!r}; expected 'sirannon' or 'postgres'")

File: benchmarks/server/driver/src/drivers/registry.ts
// Map an engine name to its driver.

import type { Config } from '../config.ts'
import type { Driver } from './driver.ts'
import { PostgresDriver } from './postgres.ts'
import { SirannonDriver } from './sirannon.ts'

export function buildDriver(engine: string, config: Config, durability: string): Driver {
  if (engine === 'sirannon') {
    return new SirannonDriver(config.sirannon.baseUrl, config.sirannon.databaseId, durability, config.requestTimeoutMs)
  }
  if (engine === 'postgres') {
    return new PostgresDriver(config.postgres, durability)
  }
  throw new Error(`unknown engine ${JSON.stringify(engine)}; expected 'sirannon' or 'postgres'`)
}
